### ISSUE TITLE
Demo: PyTorch-style tensor program in ll-lang → Python

### ISSUE THREAD

**Issue by Neftedollar** [Neftedollar/ll-lang#153]
## Goal

Build a simple PyTorch-style demo in ll-lang that compiles via \`--target py\` and uses the PyTorch Python library.

## Why

- **Demo for Python/ML developers** — largest LLM user segment
- Shows ll-lang bridging into the scientific Python ecosystem
- Exercises: numeric types, opaque types, Python module interop

## Scope

A minimal but real neural network example:
- Define a tiny MLP (2 layers, ReLU)
- Load toy dataset (or hardcode tensors)
- Forward pass
- Loss computation
- One training step with autograd

## Files

- \`examples/pytorch-mlp/src/Main.lll\`
- \`examples/pytorch-mlp/requirements.txt\` (torch)
- \`examples/pytorch-mlp/README.md\`

## Approach

Declare PyTorch bindings as externals:

\`\`\`lll
opaque Tensor
opaque Module

external torchTensor(data List[Float]) Tensor
external tensorMatmul(a Tensor)(b Tensor) Tensor
external tensorRelu(t Tensor) Tensor
external tensorBackward(t Tensor) Int
external torchLinear(inDim Int)(outDim Int) Module
external moduleForward(m Module)(x Tensor) Tensor
\`\`\`

Python output imports torch:
\`\`\`py
import torch
import torch.nn as nn
def tensorMatmul(a): return lambda b: a @ b
def tensorRelu(t): return torch.relu(t)
# ...
\`\`\`

Shim lives in \`sdks/Platform.Python.SDK/\`.

## Acceptance

- [ ] \`lllc build --target py examples/pytorch-mlp/src/Main.lll\` produces .py
- [ ] \`pip install torch && python examples/pytorch-mlp/src/Main.py\` runs forward + backward pass
- [ ] Prints loss before/after one gradient step (loss decreases)
- [ ] README with build & run

## Labels

demo, devrel, python-target, ml

**Comment by Neftedollar** [comment]
Demo landed in commit 0f196dd. MLP forward+loss in pure ll-lang, torch companion cross-checks to 1e-7 tolerance + runs backward pass with printed gradients. FFI limit prevented embedding torch calls directly — see #155.
